exchanges/deribit/deribit_test.go

```diff
@@ -0,0 +1,3036 @@
+	}
+	d.Websocket = sharedtestvalues.NewTestWebsocket()
+	err = d.Setup(exchCfg)
+	if err != nil {
+		log.Fatal("Deribit setup error", err)
+	}
+	if useTestNet {
+		for k, v := range d.API.Endpoints.GetURLMap() {
+			v = strings.Replace(v, "www.deribit.com", "test.deribit.com", 1)
+			err = d.API.Endpoints.SetRunning(k, v)
+			if err != nil {
+				log.Fatal(err)
+			}
+		}
+	}
+	d.Websocket.DataHandler = sharedtestvalues.GetWebsocketInterfaceChannelOverride()
+	d.Websocket.TrafficAlert = sharedtestvalues.GetWebsocketStructChannelOverride()
+	fetchTradablePairChan = make(chan struct{})
+	instantiateTradablePairs()
+	setupWs()
+	os.Exit(m.Run())
+}
+
+func TestUpdateTicker(t *testing.T) {
+	t.Parallel()
+	_, err := d.UpdateTicker(context.Background(), spotTradablePair, asset.Spot)
+	assert.NoError(t, err)
+	_, err = d.UpdateTicker(context.Background(), futuresTradablePair, asset.Futures)
+	assert.NoError(t, err)
+	sleepUntilTradablePairsUpdated()
+	_, err = d.UpdateTicker(context.Background(), optionsTradablePair, asset.Options)
+	assert.NoError(t, err)
+	_, err = d.UpdateTicker(context.Background(), optionComboTradablePair, asset.OptionCombo)
+	assert.NoError(t, err)
+	_, err = d.UpdateTicker(context.Background(), futureComboTradablePair, asset.FutureCombo)
+	assert.NoError(t, err)
+	_, err = d.UpdateTicker(context.Background(), currency.Pair{}, asset.Margin)
+	assert.Falsef(t, err != nil && !errors.Is(err, asset.ErrNotSupported), "expected: %v, received %v", asset.ErrNotSupported, err)
+}
+
+func TestUpdateOrderbook(t *testing.T) {
+	t.Parallel()
+	_, err := d.UpdateOrderbook(context.Background(), spotTradablePair, asset.Spot)
+	require.NoError(t, err)
+	_, err = d.UpdateOrderbook(context.Background(), futuresTradablePair, asset.Futures)
+	assert.NoError(t, err)
+	sleepUntilTradablePairsUpdated()
+	_, err = d.UpdateOrderbook(context.Background(), optionsTradablePair, asset.Options)
+	assert.NoError(t, err)
+	_, err = d.UpdateOrderbook(context.Background(), futureComboTradablePair, asset.FutureCombo)
+	assert.NoError(t, err)
+	_, err = d.UpdateOrderbook(context.Background(), optionComboTradablePair, asset.OptionCombo)
+	assert.NoError(t, err)
+}
+
+func TestGetHistoricTrades(t *testing.T) {
+	t.Parallel()
+	_, err := d.GetHistoricTrades(context.Background(), futuresTradablePair, asset.Futures, time.Now().Add(-time.Minute*10), time.Now())
+	assert.NoError(t, err)
+	_, err = d.GetHistoricTrades(context.Background(), spotTradablePair, asset.Spot, time.Now().Add(-time.Minute*10), time.Now())
+	assert.NoError(t, err)
+}
+
+func TestFetchRecentTrades(t *testing.T) {
+	t.Parallel()
+	_, err := d.GetRecentTrades(context.Background(), futuresTradablePair, asset.Futures)
+	assert.NoError(t, err)
+	_, err = d.GetRecentTrades(context.Background(), spotTradablePair, asset.Spot)
+	assert.NoError(t, err)
+	sleepUntilTradablePairsUpdated()
+	_, err = d.GetRecentTrades(context.Background(), optionsTradablePair, asset.Options)
```

Comment:
So in the [build linked here](https://github.com/thrasher-corp/gocryptotrader/actions/runs/8554867603/job/23441069586?pr=1082#step:8:958), we get a few errors about `"Invalid params","data":{"param":"instrument_name","reason":"wrong format"`
Do you think you could look into why `options` asset currency is causing this issue? I ask to prevent us from having tests break randomly based on what `sleepUntilTradablePairsUpdated` selects as an `options` currency. I have only seen this for `options` asset type

Reply:
I couldn't reproduce these errors. Could you help me do so? I have made updates to handle option pairs when making an outgoing request with the `d.optionPairToString`

Reply:
Yeah it seems a bit random depending on what option currency gets picked 😄 I'll try and find a specific instance of a bad one with logging

Reply:
I think the error was on the regexp I used when converting the options pair to string. I have made a slight update ... hopefully it won't error again.